core/http/src/mtls.rs

```diff
@@ -0,0 +1,19 @@
+use tls::Certificate;
+
+#[derive(Debug)]
+pub struct MutualTlsUser {
```

Comment:
This will need docs. Also, we need nice accessor methods for all of the things that will be commonly desired.